pkg/provider/azure_storageaccount.go

```diff
@@ -262,6 +274,35 @@ func (az *Cloud) EnsureStorageAccount(accountOptions *AccountOptions, genAccount
 					return "", "", err
 				}
 			}
+
+			if accountOptions.CreatePrivateEndpoint {
+				// Create private endpoint
+				privateEndpointName := accountName + "-PrivateEndpoint"
+				err := az.createPrivateEndpoint(ctx, accountName, privateEndpointName)
+				if err != nil {
+					return "", "", fmt.Errorf("Failed to create private endpoint for storage account(%s), resourceGroup(%s), error: %v", accountName, az.ResourceGroup, err)
+				}
+
+				// Create DNS zone
+				err = az.createPrivateDNSZone(ctx)
+				if err != nil {
+					return "", "", fmt.Errorf("Failed to create private DNS zone(%s) in resourceGroup(%s), error: %v", PrivateDnsZoneName, az.ResourceGroup, err)
+				}
+
+				// Create virtual link to the zone private DNS zone
+				vNetLinkName := accountName + "-VnetLink"
+				err = az.createVNetLink(ctx, vNetLinkName)
+				if err != nil {
+					return "", "", fmt.Errorf("Failed to create virtual link for vnet(%s) and DNS Zone(%s) in resourceGroup(%s), error: %v", az.VnetName, PrivateDnsZoneName, az.ResourceGroup, err)
+				}
+
+				// Create dns zone group
+				dnsZoneGroupName := accountName + "-dnsZoneGroup"
```

Comment:
np: use consistent casing in suffixes - here it starts with a lowercase letter where the others start with uppercase. Should we stick to all lowercase letters in any case? Many Azure resource names require lowercase.

Reply:
actually just use `accountName` is ok, no need to add a suffix

Reply:
@andyzhangx Wouldn't it be confusing to have the same name for all these different resources?